### PR TITLE
add reaper

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -39,4 +41,5 @@ COPY rancher /usr/bin/
 ENV CATTLE_AGENT_IMAGE rancher/agent:${VERSION}
 ENV CATTLE_SERVER_IMAGE rancher/server
 VOLUME /var/lib/rancher
-ENTRYPOINT ["rancher", "--http-listen-port=80", "--https-listen-port=443"]
+ENTRYPOINT ["tini", "--"]
+CMD ["rancher", "--http-listen-port=80", "--https-listen-port=443"]
```
Comment (ibuildthecloud) [inline package/Dockerfile]: This needs to be `ENTRYPOINT ["tini", "--", "rancher", "--http-listen-port=80", "--https-listen-port=443"]`

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/13858

File: app/app.go [hunk 2]
```diff
@@ -140,3 +149,63 @@ func addData(management *config.ManagementContext, cfg Config) error {
...
+	}
+
+	return 0
+}
+
+func reaper() {
+	reap := make([]string, 0, 5)
+
+	for {
+		time.Sleep(1 * time.Second)
```
Comment (cjellick) [inline app/app.go]: Why do we need to do this every second?

File: app/app.go [hunk 3]
```diff
@@ -2,6 +2,13 @@ package app
 
 import (
 	"context"
+	"io/ioutil"
+	"strings"
+	"time"
+
```
Comment (cjellick) [inline app/app.go]: no need for this newline

File: app/app.go (hunk 2 above)
Comment (StrongMonkey) [reply app/app.go]: :) because i copied the whole method darren gave me

File: app/app.go [hunk 4]
```diff
@@ -31,6 +39,7 @@ type Config struct {
 }
 
 func buildScaledContext(ctx context.Context, kubeConfig rest.Config, cfg *Config) (*config.ScaledContext, *clustermanager.Manager, error) {
+	NewDaemon()
```
Comment (cjellick) [inline app/app.go]: What is the motivation for doing thi sin the buildScaledContext function? It does not seem like the right place for this. Woudl make more sense in the function that calls buildScaledContext

File: app/app.go (hunk 2 above)
Comment (StrongMonkey) [reply app/app.go]: To be discussed...

File: app/app.go [hunk 5]
```diff
@@ -140,3 +149,63 @@ func addData(management *config.ManagementContext, cfg Config) error {
...
+			if getZombiePid(file) > 0 {
+				reap = append(reap, file)
+			}
+		}
+	}
+}
+
+// NewDaemon sets up everything for the daemon to be able to service
+// requests from the webserver.
+func NewDaemon() {
```
Comment (cjellick) [inline app/app.go]: NewDaemon is not a good name. Can you rename it something more descriptive?
Also, only export it if it is being called outside of this package

**Comment by StrongMonkey** [comment]
close in favor of https://github.com/rancher/rancher/pull/14515

**Pull request closed without merging**